jsk_robot_common/jsk_robot_startup/euslisp/smach-utils.l

```diff
@@ -0,0 +1,15 @@
+(defmacro def-state-func (name args &rest body)
+  (let ((userdata
+          (cond ((eq (car args) '&optional)
+                 (cond
+                   ((listp (cadr args))
+                    (if (listp (caadr args))
+                      (cadr (caadr args))
+                      (caadr args)))
+                   (t (cadr args))))
+                ((eq (car args) '&rest) `(car ,(cadr args)))
+                (t (car args)))))
```

Comment:
Maybe also `((null args) (error "Function must have the userdata parameter"))` ?